docs/blueprints/referral-manager-v1.md

```diff
@@ -0,0 +1,236 @@
+
+
+_Referral Manager CLI_
+
+1. `referral manager start` initiates the invitation token generation process. This process begins by checking whether there are existing `eligibleUsers` in memory in Referral Manager.
+    - If `eligibleUsers` is empty, the Referral Manager queries `users` table to get users with 0 remaining invites.
+    - If `eligibleUsers` is not empty, it means one of two things:
+        * Someone else is using the Referral Manager at the same time and has not generated/sent tokens yet.
+        * The CLI was terminated between the get `eligibleUsers` step and the generate/send tokens step.
+    
+        Either one of the cases, We will return an error and display `Referral token generation already in progress or canceled early. Do you want to continue anyway?`. If they decide to continue, we call `ClearUsers` to sets eligibleUsers to empty and the Referral Manager queries `users` table to get users with 0 remaining invites. 
+2. The Referral Manager returns the number of users to the CLI.
+3. The CLI receives the number of users and displays: `How many invitation do you want to generate for each user?`
+4. After the user enters the number of invites to generate, the CLI displays: `Generating X amount of invitation tokens for Y amount of users. Yes/No.`
+   - If the input is `Yes`, it will generate x amount of invitation tokens per user based on the input from CLI and save them into `tokens` table.
+   - The CLI will call `ClearUsers` endpoint to set `eligibleUser` to be empty if the input is `No`.
+
+_Referral Link Redemption_
+
+1. User Alice tries to register a new account through a referral link, which triggers redeemed satellite to verify invitation token using Referral manager.
+2. Referral Manager checks the status of the token:
+    - if the token exists in `tokens` table:
+        - if it is not redeemed, Referral Manager sends back a success response to the redeemed satellite and mark the token as redeemed in the Referral Manager's database
+        - if it token is already redeemed, Referral Manager sends back a `invalid token` response to the redeemed satellite.
+        
+    - if the token doesn't exist in `tokens` table, Referral Manager sends back a `invalid token` response to the redeemed satellite.
+3. Redeemed satellite receives the response, which:
+    - if it is a success, the satellite will proceed with the account creation and
+    then send a request to the Referral Manager, which: 
+        - updates the `tokens` table to store the newly created user ID in `redeemedID` column for that particular referral token.
+        - updates the `users` table to store the newly created user ID and the satellite URL the user belongs to.
+    - if it is an invalid token, the satellite will display a proper message in the UI.
+    
+## Rationale
+
+We will create a `inspector` port that's only listening on `localhost` so CLI could be only used by Referral Manager operators.
```

Comment:
Inspector port doesn't sound quite right. Something like `admin`, `cli` might be better.

Reply:
Fixed.